CaImAn/caiman_main.py

```diff
@@ -0,0 +1,316 @@
+                operationCount = str(len(operations))
+                for operation in operations:
+                    operationAttrs = utils.load_attributes(f, f"{vpath}/{operation}")
+                    if utils.merge_params(params_doric, params_source) == operationAttrs:
+                        if(len(operation) == len(cm_defs.DoricFile.Group.ROISIGNALS)):
+                            operationCount = ""
+                        else:
+                            operationCount = operation[-1]
+
+                        break
+
+        params_doric[defs.DoricFile.Attribute.Group.OPERATIONS] += operationCount
+
+        print(cm_defs.Messages.SAVE_ROI_SIG, flush = True)
+        pathROIs          = f"{vpath}/{cm_defs.DoricFile.Group.ROISIGNALS+operationCount}"
+        pathROIs_vdataset = f"{pathROIs}/{vdataset}"
+        utils.save_roi_signals(C, A, time_, f, pathROIs_vdataset, attrs_add={"RangeMin": 0, "RangeMax": 0, "Unit": "Intensity"})
+        utils.print_group_path_for_DANSE(pathROIs_vdataset)
+        utils.save_attributes(utils.merge_params(params_doric, params_source), f, pathROIs)
+
+        if saveimages:
+            print(cm_defs.Messages.SAVE_IMAGES, flush = True)
+            pathImages          = f"{vpath}/{cm_defs.DoricFile.Group.IMAGES+operationCount}"
+            pathImages_vdataset = f"{pathImages}/{vdataset}"
+            utils.save_images(AC, time_, f, pathImages_vdataset, bit_count=bit_count, qt_format=qt_format, username=username)
+            utils.print_group_path_for_DANSE(pathImages_vdataset)
+            utils.save_attributes(utils.merge_params(params_doric, params_source, params_doric[defs.DoricFile.Attribute.Group.OPERATIONS] + "(Images)"), f, pathImages)
+
+        if saveresiduals:
+            print(cm_defs.Messages.SAVE_RES_IMAGES, flush = True)
+            pathResiduals          = f"{vpath}/{cm_defs.DoricFile.Group.RESIDUALS+operationCount}"
+            pathResiduals_vdataset = f"{pathResiduals}/{vdataset}"
+            utils.save_images(res, time_, f, pathResiduals_vdataset, bit_count=bit_count, qt_format=qt_format, username=username)
+            utils.print_group_path_for_DANSE(pathResiduals_vdataset)
+            utils.save_attributes(utils.merge_params(params_doric, params_source, params_doric[defs.DoricFile.Attribute.Group.OPERATIONS] + "(Residuals)"), f, pathResiduals)
+
+        if savespikes:
+            print(cm_defs.Messages.SAVE_SPIKES, flush = True)
+            pathSpikes          = f"{vpath}/{cm_defs.DoricFile.Group.SPIKES+operationCount}"
+            pathSpikes_vdataset = f"{pathSpikes}/{vdataset}"
+            utils.save_signals(S > 0, time_, f, pathSpikes_vdataset, names, roiUsernames, range_min=0, range_max=1)
+            utils.print_group_path_for_DANSE(pathSpikes_vdataset)
+            utils.save_attributes(utils.merge_params(params_doric, params_source), f, pathSpikes)
+
+    print(cm_defs.Messages.SAVE_TO.format(path = vname), flush = True)
+
+
+def set_advanced_parameters(
```

Comment:
I think, in minian, this is part of the caiman_parameters.py, I would do similar here as well

Reply:
the class `params.CNMFParams` can be in the caiman_parameters.py